### PR TITLE
Search every overmap for npcs to continue as once you've died

### PR DESCRIPTION
**Pull request by perryprog**
#### Summary
None

#### Purpose of change
This amends commit 8f900e5, as it turns out that was an accidental revert of PR #56220, which was a bugfix.

#### Describe the solution
To correctly hide the "Continue as one of your followers" message, every overmap and each npc of every overmap should be iterated through, which is what this commit does by way of adding an EOC condition based off of the already existing `npc_allies` one.

#### Describe alternatives you've considered
To use `game::get_follower_list`, which is what `avatar::control_npc_menu` uses. While I originally didn't realize that method existed, I think it's better _not_ to use it in this one case: I noticed a few instances during testing where I was getting the option to take control of my dead self. (Specifically: as a new player I would get a follower, die, control the follower, die, and get the option to control the original player where if I did I would immediately be put into the "Watch the last movements of your life" dialog. For whatever reason, I could never get that to occur using this PR's current patch.

(Note: I suspect I'm missing something here because in one of the tests I did I was controlling an NPC with 1 NPC follower and the player having just died. In that case, I should've seen the take-control menu as the count my code gave would've been 1... this didn't happen, however)

#### Testing
I tested just about every circumstance I could think of (multiple followers, being in the overmap and out of the overmap of the player's followers, dying with no followers) and I didn't notice any unexpected behavior. I still suspect there could still be something up as described above, though since there's no current buggy behavior I can get to happen, I don't think it's too much of an issue.

### CODE REVIEW

**Comment by PatrikLundell** [comment]
Have you tested what happens if your only follower is assigned to a base camp or is currently on a mission (which could be a camp one or one for the Tacoma commune), as well as busy doing some ordinary task (clean up, chop trunks into planks, dismantle vehicles etc.)?
That's not really related to the fix, but to the underlying functionality.

**Comment by perryprog** [comment]
> Have you tested what happens if your only follower is assigned to a base camp or is currently on a mission (which could be a camp one or one for the Tacoma commune), as well as busy doing some ordinary task (clean up, chop trunks into planks, dismantle vehicles etc.)?
> That's not really related to the fix, but to the underlying functionality.

I just checked—very surprisingly, it works fine for an NPC assigned to a basecamp mission. They sort of just re-appear where they disappeared from if you choose to possess one. I haven't tested out any other combinations, though.

File: src/condition.cpp [hunk 1]
```diff
@@ -729,6 +729,26 @@ void conditional_t<T>::set_npc_allies( const JsonObject &jo )
...
 
+
+template<class T>
+void conditional_t<T>::set_npc_allies_global( const JsonObject &jo )
+{
+    int_or_var iov = get_int_or_var( jo, "npc_allies_global" );
+
+    condition = [iov]( const T & d ) {
+        const auto all_npcs = overmap_buffer.get_overmap_npcs();
+        std::vector<shared_ptr_fast<npc>> filtered_npcs{};
```
Comment (BrettDong) [inline src/condition.cpp:740]: Is this used anywhere?

Comment (perryprog) [reply src/condition.cpp]: Whoops, no it's not. That was from an earlier thing I was trying.

**Pull request merged**